app/src/main/java/org/fossasia/openevent/general/favorite/FavoriteFragment.kt

```diff
@@ -70,6 +81,45 @@ class FavoriteFragment : Fragment() {
                 showEmptyMessage(favoriteEventsRecyclerAdapter.itemCount)
             }
         }
+        val calendar = Calendar.getInstance()
+        setCurrentDate(calendar)
+        rootView.searchToday.setOnClickListener {
+            searchTimeViewModel.saveDate(EventUtils.getSimpleFormattedDate(calendar.time))
+            calendar.add(Calendar.DATE, 1)
+            searchTimeViewModel.saveNextDate(EventUtils.getSimpleFormattedDate(calendar.time))
+            searchTimeViewModel.saveDate(TODAY)
+            redirectToSearch()
+        }
+        rootView.searchTomorrow.setOnClickListener {
+            calendar.add(Calendar.DATE, 1)
+            searchTimeViewModel.saveNextDate(EventUtils.getSimpleFormattedDate(calendar.time))
+            calendar.add(Calendar.DATE, 1)
+            searchTimeViewModel.saveNextToNextDate(EventUtils.getSimpleFormattedDate(calendar.time))
+            searchTimeViewModel.saveDate(TOMORROW)
+            redirectToSearch()
+        }
+        rootView.searchWeekend.setOnClickListener {
+            val today = calendar.get(Calendar.DAY_OF_WEEK)
+            if (today != Calendar.SATURDAY) {
+                val offset = Calendar.SATURDAY - today
+                calendar.add(Calendar.DATE, offset)
+            }
+            searchTimeViewModel.saveWeekendDate(EventUtils.getSimpleFormattedDate(calendar.time))
+            calendar.add(Calendar.DATE, 1)
+            searchTimeViewModel.saveNextToWeekendDate(EventUtils.getSimpleFormattedDate(calendar.time))
+            searchTimeViewModel.saveDate(THIS_WEEKEND)
+            redirectToSearch()
+        }
+        rootView.searchNextMonth.setOnClickListener {
+            val today = calendar.get(Calendar.DAY_OF_MONTH)
+            val offset = 30 - today
+            calendar.add(Calendar.DATE, offset)
+            searchTimeViewModel.saveNextMonth(EventUtils.getSimpleFormattedDate(calendar.time))
+            calendar.add(Calendar.MONTH, 1)
+            searchTimeViewModel.saveNextToNextMonth(EventUtils.getSimpleFormattedDate(calendar.time))
+            searchTimeViewModel.saveDate(NEXT_MONTH)
+            redirectToSearch()
+        }
```

Comment:
All logic in view model

Reply:
Implemented the onClick logic in the ViewModel.